dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java

```diff
@@ -0,0 +1,354 @@
+            if (certificateChain.isEmpty()) {
+                throw new CertificateException(
+                        "Certificate file does not contain any certificates: "
+                                + certificateChainFile);
+            }
+
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(null, null);
+            keyStore.setKeyEntry("key", key, keyPassword.orElse("").toCharArray(),
+                    certificateChain.stream().toArray(
+                            Certificate[]::new));
+            return keyStore;
+        }
+
+        private static PKCS8EncodedKeySpec readPrivateKey(File keyFile,
+                Optional<String> keyPassword)
+                throws IOException, GeneralSecurityException {
+            String content = readFile(keyFile);
+
+            Matcher matcher = KEY_PATTERN.matcher(content);
+            if (!matcher.find()) {
+                throw new KeyStoreException("found no private key: " + keyFile);
+            }
+            byte[] encodedKey = base64Decode(matcher.group(1));
+
+            if (!keyPassword.isPresent()) {
+                return new PKCS8EncodedKeySpec(encodedKey);
+            }
+
+            EncryptedPrivateKeyInfo encryptedPrivateKeyInfo = new EncryptedPrivateKeyInfo(
+                    encodedKey);
+            SecretKeyFactory keyFactory = SecretKeyFactory
+                    .getInstance(encryptedPrivateKeyInfo.getAlgName());
+            SecretKey secretKey = keyFactory
+                    .generateSecret(new PBEKeySpec(keyPassword.get().toCharArray()));
+
+            Cipher cipher = Cipher.getInstance(encryptedPrivateKeyInfo.getAlgName());
+            cipher.init(DECRYPT_MODE, secretKey, encryptedPrivateKeyInfo.getAlgParameters());
+
+            return encryptedPrivateKeyInfo.getKeySpec(cipher);
+        }
+
+        private static List<X509Certificate> readCertificateChain(File certificateChainFile)
+                throws IOException, GeneralSecurityException {
+            String contents = readFile(certificateChainFile);
+
+            Matcher matcher = CERT_PATTERN.matcher(contents);
+            CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+            List<X509Certificate> certificates = new ArrayList<>();
+
+            int start = 0;
+            while (matcher.find(start)) {
+                byte[] buffer = base64Decode(matcher.group(1));
+                certificates.add((X509Certificate) certificateFactory
+                        .generateCertificate(new ByteArrayInputStream(buffer)));
+                start = matcher.end();
+            }
+
+            return certificates;
+        }
+
+        private static byte[] base64Decode(String base64) {
+            return Base64.getMimeDecoder().decode(base64.getBytes(US_ASCII));
+        }
+
+        private static String readFile(File file)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'file' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=3712629)